custom/icds_reports/reports/awc_reports.py

```diff
@@ -1040,10 +1040,11 @@ def base_data(row_data):
     return config
 
 
-@quickcache(['case_id'], timeout=30 * 60)
-def get_beneficiary_details(case_id):
-    data = ChildHealthMonthlyView.objects.filter(
-        case_id=case_id
+@quickcache(['case_id', 'awc_id'], timeout=30 * 60)
+def get_beneficiary_details(case_id, awc_id):
+    data = ChildHealthMonthly.objects.filter(
```

Comment:
I think that you can move this back to `ChildHealthMonthlyView` as that wasn't the cause for slowness